js/models/purchase/Order.js

```diff
@@ -57,8 +59,19 @@ export default class extends BaseModel {
       addError('items', 'At least one item is required.');
     }
 
-    if (this.shippable && !attrs.shipTo && !attrs.countryCode) {
-      addError('shipping', app.polyglot.t('orderModelErrors.missingAddress'));
+    const c = attrs.paymentCoin;
+    if (!(c && typeof c === 'string' && isSupportedWalletCur(c))) {
+      addError('paymentCoin', app.polyglot.t('orderModelErrors.paymentCoinInvalid'));
+    }
+
+    if (this.shippable) {
+      if (!attrs.shipTo || !(typeof attrs.shipTo === 'string')) {
+        addError('shipping', app.polyglot.t('orderModelErrors.missingAddress'));
+      }
+
+      if (!attrs.countryCode || !(typeof attrs.countryCode === 'string')) {
+        addError('shipping', app.polyglot.t('orderModelErrors.missingAddress'));
+      }
```

Comment:
There was a bug in the shipping validation where it only failed validation if both the shipTo and countryCode were false.